components/tidb_query/src/rpn_expr/impl_string.rs

```diff
@@ -238,6 +239,33 @@ pub fn locate_binary_3_args(
     }
 }
 
+#[rpn_fn(raw_varg, min_args = 2, extra_validator = elt_validator)]
+#[inline]
+pub fn elt(raw_args: &[ScalarValueRef]) -> Result<Option<Bytes>> {
+    assert!(raw_args.len() >= 2);
+    let index = raw_args[0].as_int();
+    Ok(match *index {
+        None => None,
+        Some(i) => {
+            if i <= 0 || i + 1 > raw_args.len() as i64 {
+                return Ok(None);
+            }
+            raw_args[i as usize].as_bytes().to_owned()
+        }
+    })
+}
+
+/// validate the arguments are `(&Option<Int>, &[&Option<Bytes>)])`
+fn elt_validator(expr: &tipb::Expr) -> Result<()> {
+    let children = expr.get_children();
+    assert!(children.len() >= 2);
```

Comment:
It's better to return an error instead of panicking to avoid illegal push down parameters make TiKV crash..

Reply:
Hi, I think I should remove this check because the `min_args` check already exists. What do you think?

Reply:
I think so.

Reply:
updated.

Reply:
The purpose of `assert!(children.len() >= X);` in other validators is to hint the compiler about the boundary so that it won't check the bound for accessing `[0]`, `[1]`. `min_args` already checks the children length so this assertion asserted this fact.

Reply:
`debug_assert!` looks better?

Reply:
@ty666 I am sorry to misleading you, according to @breeswish 's comment (https://github.com/tikv/tikv/pull/6062#discussion_r351072379), it's better to `assert`  here.

Reply:
I mean it might be better to use `debug_assert!`  instead of `assert!`.

Reply:
You may keep using assert! to be unified with other validators.

Reply:
https://doc.rust-lang.org/stable/core/macro.assert.html#uses

> Assertions are always checked in both debug and release builds, and cannot be disabled. See debug_assert! for assertions that are not enabled in release builds by default.

Reply:
> You may keep using assert! to be unified with other validators.

Updated. thanks.